src/validation.h

```diff
@@ -684,7 +684,7 @@ class CChainState
 
     /** Replay blocks that aren't fully applied to the database. */
     bool ReplayBlocks(const CChainParams& params);
-    bool RewindBlockIndex(const CChainParams& params) LOCKS_EXCLUDED(cs_main);
+    [[nodiscard]] bool NeedsRedownload(const CChainParams& params) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```

Comment:
Add a doxygen comment? `/** Check if chain state needs to be redownloaded due to lack of witness data */` ?

Should be a `const` method, as far as I can see.

Reply:
Done and done.